server/server.js

```diff
@@ -1,15 +1,122 @@
 const express = require("express");
 const app = express();
+const cors = require("cors");
+app.use(cors());
 const port = process.env.PORT || 5000;
+app.use(express.urlencoded());
+app.use(express.json());
+app.listen(port, function () {
+  console.log("Server is listening!");
+});
+
+const { Pool } = require("pg");
+const pool = new Pool({
+  connectionString:
+    "postgres://klzydhijtaanrz:ad95a4e271095b78b9c1b476f66f6524b24ad7f99185033b6487395f3d30d404@ec2-52-54-38-229.compute-1.amazonaws.com:5432/d94341lme3orfj",
+  ssl: { rejectUnauthorized: false },
+  user: "klzydhijtaanrz",
+  host: "ec2-52-54-38-229.compute-1.amazonaws.com",
+  database: "d94341lme3orfj",
+  password: "ad95a4e271095b78b9c1b476f66f6524b24ad7f99185033b6487395f3d30d404",
+  port: 5432,
+});
+//get all videos
+
+pool.connect();
+app.get("/", function (req, res) {
+  res.header("Access-Control-Allow-Origin", "*");
+  res.header("Access-Control-Allow-Credentials", true);
+  res.header("Access-Control-Allow-Methods", "GET,PUT,POST,DELETE,OPTIONS");
+  res.header(
+    "Access-Control-Allow-Headers",
+    "Origin,X-Requested-With,Content-Type,Accept,content-type,application/json"
+  );
+
+  let order = req.query.order;
+  let query;
+  if (order === "asc")
+    query = "SELECT * FROM videos order by rating";
+  else query = "SELECT * FROM videos order by rating desc";
+
+  pool
+    .query(query)
+    .then((result) => {
+      if (result.rowCount > 0)
+        res.json(result.rows);
+      else res.send("no videos available");
+    })
+    .catch((e) => res.status(500).send("server error"));
+});
+
+app.get("/:id", (req, res) => {
+  let videoId = Number(req.params.id);
+
+  const query = "select * from videos where id=$1";
+
+  pool
+    .query(query, [videoId])
+    .then((result) => {
+      if (result.rowCount > 0) res.status(200).json(result.rows);
+      else
+        res.status(404).json({
+          result: "failure",
+          message: "Video could not be found",
+        });
+    })
+    .catch((e) => res.status(500).send("server error"));
```

Comment:
Good work in identifying client side error and server side error, and having different response codes for each.